netsim-ffi/src/lib.rs

```diff
@@ -86,8 +87,20 @@ pub unsafe extern "C" fn netsim_context_open(
     if context.is_null() || output.is_null() {
         SimError::NullPointerArgument
     } else {
-        // TODO
+        let mut context = Box::from_raw(context);
```

Comment:
you can't put it in a box or it will be released from memory when the function returns.

Instead use `as_mut()` or `as_ref()` depending if you need the mutability of the Context.

See the `netsim_socket_send_to` where I used similar for the `SimSocket`.